extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -194,7 +195,7 @@ public boolean start()
             public void run()
             {
               while (!executorService.isShutdown()) {
-                final ConsumerConnector consumerConnector = buildConnector(kafkaProperties);
+                consumerConnector = buildConnector(kafkaProperties);
```

Comment:
Ah, okay, nevermind, I see what you're saying. The check for isShutdown happens again after creating the connector and that prevents the race. I missed that check when I was looking at the diff, thanks for explaining it.